source/as-promise.ts

```diff
@@ -62,22 +67,25 @@ export default function asPromise(options: Options) {
 			response.body = data;
 
 			try {
-				for (const [index, hook] of options.hooks!.afterResponse!.entries()) {
+				for (const [index, hook] of options.hooks.afterResponse.entries()) {
 					// eslint-disable-next-line no-await-in-loop
 					response = await hook(response, updatedOptions => {
```

Comment:
```suggestion
					response = await hook(response, (updatedOptions: NormalizedOptions) => {
```

so we won't have to do `updatedOptions as NormalizedOptions`.

Reply:
We have to do that cast since the type for `updatedOptions` in that context is DeepPartial\<NormalizedOptions\>. I think I can make merge return the correct type.